frontend/src/store/cats.store.ts

```diff
@@ -1,35 +1,47 @@
-import { AxiosResponse } from "axios"
+import axios, { AxiosResponse } from "axios"
 import constate from "constate"
 import debounce from "debounce-promise"
-import { useState } from "react"
+import { useCallback, useRef, useState } from "react"
 import type { ICat } from "../types/common.types"
-import {createCat as createCatService, getCats as getCatsService}  from "../api/cats.service"
+import { createCat as createCatService, getCats as getCatsService } from "../api/cats.service"
 
-const getCatsDebounced = debounce(getCatsService, 300)
-
-const useCatsStore = ({initialValue = []}) => {
+const useCatsStore = ({ initialValue = [] }) => {
     const [cats, setCats] = useState<Array<ICat>>(initialValue)
-    const createCat = async (cat: Partial<ICat>) => {
+    const controllerRef = useRef<AbortController | null>();
+
+    const getCatsDebounced = useCallback(debounce((searchText: string) => {
+        if (controllerRef.current) {
+            controllerRef.current.abort();
+            controllerRef.current = null;
+        }
+        controllerRef.current = new AbortController();
+        return getCatsService(controllerRef.current.signal, searchText)
+    }, 300), [])
+
+    const getCats = useCallback(async (searchText = "") => {
         try {
-            const {data: newCat} = await createCatService(cat) as AxiosResponse<ICat>;          
-            setCats([...cats, newCat])
-        } catch (error) { 
+            const { data: cats } = await getCatsDebounced(searchText) as AxiosResponse<ICat[]>;
+            controllerRef.current = null;
+            setCats(cats)
+        } catch (error) {
+            if (axios.isCancel(error)) {
+                return
+            }
             throw new Error(String(error))
         }
-    }
+    }, [])
 
-    const getCats = async (searchText = "") => {
+    const createCat = useCallback(async (cat: Partial<ICat>) => {
         try {
-          const invokeFunction = searchText ? getCatsDebounced : getCatsService;
-            const {data: cats} = await invokeFunction(searchText) as AxiosResponse<ICat[]>;          
-            setCats(cats)
-        } catch (error) { 
+            const { data: newCat } = await createCatService(cat) as AxiosResponse<ICat>;
```

Comment:
instead of using `as AxiosResponse<ICat>` it should already be the stated type in the service 
```tsx
const createCat = (catOptions: TCatOptions): AxiosResponse<ICat> => {
```